DynamicFlightStorageSimulation/SimulationConsumer.cs

```diff
@@ -130,6 +130,20 @@ await _simulationEventBus.PublishSystemMessage(new SystemMessage()
                     await _consumerLogger.PersistDataAsync(_simulationEventBus.CurrentExperimentId, ClientId).ConfigureAwait(false);
                     _consumerLogger.ResetLogger();
                     break;
+                case SystemMessage.SystemMessageType.AbortExperiment:
+                    _logger?.LogWarning("Consumer aborting experiment");
+                    await _simulationEventBus.ClearExchanges().ConfigureAwait(false); // Problem: queues are cleared but messages are still attempted to be read
```

Comment:
As a fix for the messages still being attempted to be read. It's a bit hacky, but introduce a `_isCancelled` member variable and set it to `true` here. Then in OnFlightReceived and OnWeatherReceived, check the variable, **send an ACK** (maybe. Depends on how RabbitMQ treats giving you a message that you then never do anything with) and return immediately if it's set. Then in  `ResetStateAsync()`, reset the variable to `false`.